enhancements/network/on-prem-configurable-lb-dns-stack.md

```diff
@@ -0,0 +1,205 @@
+
+### Standard SLO Behavior
+
+The CHNSO is expected to follow the standard behavior of SLO, additionally, since the CHNSO is applicable only to clusters running on on-prem ( 'baremetal', 'openstack', 'vsphere' or 'ovirt') infrastructure,
+It should support also the expected behaviors for "not in use" SLOs similar to the [CBO operator](https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/an-slo-for-baremetal.md#standard-slo-behaviors).
+
+### cluster-hosted-net-services-operator details
+
+The CHNSO should:
+
+- Be a new `openshift/cluster-hosted-net-services-operator` project.
+- Publish an image called `cluster-hosted-net-services-operator`.
+- Add a new `ClusterOperator` with an additional
+  `Disabled` status for non on-prem platforms.
+- Implement a controller reconciling a singleton instance representing the desired cluster hosted LB/DNS/VIP configuration.
+- Do nothing except set `Disabled=true`, `Available=true`, and
+  `Progressing=False` when the `Infrastructure` resource is a platform
+  type other than openstack, baremetal, ovirt or vsphere.
+
+#### Kubelet and CHNSO circular dependency
+
+In order to run the CHNSO components, Kubelet in each node should first join the cluster by communicating with the control plane, however the components provided by the CHNSO are responsible for enabling that.
+As a result, the below circular dependency should be addressed:
+* Kubelet can't talk to the control plane until the CHNSO has started.
+* CHNSO can't start until Kubelet can talk to the control plane.
+
+To break the circular dependency, the CHNSO should start running before bootstrap is completed in order to assure that bootstrap owns the API-VIP. In addition, a new /etc/hosts entry should be added to the nodes using MCO ignition.  This /etc/hosts entry is expected to resolve api-int hostname to the API-VIP address.
+
+#### Disaster recovery support
+
+TODO: describe the disaster recovery use case in the context of CHNSO. 
+
+#### Self hosted DNS services
+
+The self hosted DNS functionality is composed of pods running CoreDNS-mDNS in each node and a NetworkManager dispatcher script used to configure resolv.conf to point at the node's public IP address, see [self hosted DNS design](https://github.com/openshift/installer/blob/f6bb83784a835e1f4955ef41916793c0948fd4a1/docs/design/baremetal/networking-infrastructure.md#internal-dns) for more details.
+The CHNSO will be responsible only for the deployment of the CoreDNS pods while the adjustment of the resolv.conf file will remain in the MCO, in case hosted DNS services should be disabled the CoreDNS instance should be configured as a forwarding server.
+
+#### CRD definition
+
+Below is a possible option for a CRD instance of this operator where all fields are of type Enable/Disable enum:
+
+
+```yaml
+apiVersion: clusterhostednetservices.operator.io/v1alpha1
```

Comment:
isn't this something for the config.openshift.io namespace? E.g. where the infrastructure API lives as well: https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_infrastructure.crd.yaml

Reply:
Yep, could be a good place to host the operator components.
Thank you for raising this issue.

Reply:
version is still wrong